plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/model/ComplexNode.java

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright (c) 2017 Marcus Mews.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Marcus Mews - Initial API and implementation
+ */
+package org.eclipse.n4js.flowgraphs.model;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.n4js.flowgraphs.ControlFlowType;
+import org.eclipse.n4js.flowgraphs.FGUtils;
+import org.eclipse.n4js.flowgraphs.factories.CFEMapper;
+import org.eclipse.n4js.flowgraphs.factories.ListUtils;
+import org.eclipse.n4js.n4JS.ControlFlowElement;
+
+/** Represents a complete {@link ControlFlowElement}. */
+public class ComplexNode implements ControlFlowable {
+	final private ControlFlowElement container;
+	final private ControlFlowElement astElement;
+	final private Map<String, Node> nodeMap = new HashMap<>();
+
+	private Node entry, exit;
+	private RepresentingNode represent;
+
+	/** Constructor */
+	public ComplexNode(ControlFlowElement astElement) {
+		this.container = FGUtils.getCFContainer(astElement);
+		this.astElement = astElement;
+	}
+
+	/**
+	 * Adds the given nodes to the set of internal successors, in the order from left to right with the type
+	 * {@literal ControlFlowType.Successor} Nulls are omitted. Only nodes that have been added to this
+	 * {@link ComplexNode} are allowed.
+	 */
+	public void connectInternalSucc(Node... nodes) {
+		connectInternalSucc(ControlFlowType.Successor, Arrays.asList(nodes));
+	}
+
+	/**
+	 * Adds the given nodes to the set of internal successors, in the order from left to right with the type
+	 * {@literal ControlFlowType.Successor} Nulls are omitted. Only nodes that have been added to this
+	 * {@link ComplexNode} are allowed.
+	 */
+	public void connectInternalSucc(List<Node> nodes) {
+		connectInternalSucc(ControlFlowType.Successor, nodes);
+	}
+
+	/**
+	 * Adds the given nodes to the set of internal successors, in the order from left to right with the given
+	 * {@link ControlFlowType}. Nulls are omitted. Only nodes that have been added to this {@link ComplexNode} are
+	 * allowed.
+	 */
+	public void connectInternalSucc(ControlFlowType cfType, Node... nodes) {
+		connectInternalSucc(cfType, Arrays.asList(nodes));
+	}
+
+	/**
+	 * Adds the given nodes to the set of internal successors, in the order from left to right with the given
+	 * {@link ControlFlowType}. Nulls are omitted. Only nodes that have been added to this {@link ComplexNode} are
+	 * allowed.
+	 */
+	public void connectInternalSucc(ControlFlowType cfType, List<Node> nodes) {
+		nodes = ListUtils.filterNulls(nodes);
+
+		Iterator<Node> it = nodes.iterator();
```

Comment:
Please do not use 'it' as I needed 10 seconds to recognize it is an iterator and not Xtend 'it'!

Reply:
done